drivers/can/can_kvaser_pci.c

```diff
@@ -0,0 +1,191 @@
+	.set_state_change_callback = can_sja1000_set_state_change_callback,
+	.get_core_clock = can_kvaser_pci_get_core_clock,
+	.get_max_filters = can_sja1000_get_max_filters,
+	.get_max_bitrate = can_sja1000_get_max_bitrate,
+#ifndef CONFIG_CAN_AUTO_BUS_OFF_RECOVERY
+	.recover = can_sja1000_recover,
+#endif /* !CONFIG_CAN_AUTO_BUS_OFF_RECOVERY */
+	.timing_min = CAN_SJA1000_TIMING_MIN_INITIALIZER,
+	.timing_max = CAN_SJA1000_TIMING_MAX_INITIALIZER,
+};
+
+#define CAN_KVASER_PCI_OCR                                                                         \
+	(CAN_SJA1000_OCR_OCMODE_NORMAL | CAN_SJA1000_OCR_OCTN0 | CAN_SJA1000_OCR_OCTP0 |           \
+	 CAN_SJA1000_OCR_OCTN1 | CAN_SJA1000_OCR_OCTP1)
+
+#define CAN_KVASER_PCI_CDR (CAN_SJA1000_CDR_CD_DIV2 | CAN_SJA1000_CDR_CLOCK_OFF)
+
+#define CAN_KVASER_PCI_INIT(inst)                                                                  \
+	static void can_kvaser_pci_config_func_##inst(const struct device *dev);                   \
+                                                                                                   \
+	static const struct can_kvaser_pci_config can_kvaser_pci_config_##inst = {                 \
+		.pcie_bdf = DT_INST_REG_ADDR(inst),                                                \
+		.pcie_id = DT_INST_REG_SIZE(inst),                                                 \
+		.irq_config_func = can_kvaser_pci_config_func_##inst                               \
+	};                                                                                         \
+                                                                                                   \
+	static const struct can_sja1000_config can_sja1000_config_##inst =                         \
+		CAN_SJA1000_DT_CONFIG_INST_GET(inst, &can_kvaser_pci_config_##inst,                \
+					       can_kvaser_pci_read_reg, can_kvaser_pci_write_reg,  \
+					       CAN_KVASER_PCI_OCR, CAN_KVASER_PCI_CDR);            \
+                                                                                                   \
+	static struct can_kvaser_pci_config can_kvaser_pci_data_##inst;                            \
+                                                                                                   \
+	static struct can_sja1000_data can_sja1000_data_##inst =                                   \
+		CAN_SJA1000_DATA_INITIALIZER(&can_kvaser_pci_data_##inst);                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(inst, can_kvaser_pci_init, NULL, &can_sja1000_data_##inst,           \
+			      &can_sja1000_config_##inst, POST_KERNEL, CONFIG_CAN_INIT_PRIORITY,   \
+			      &can_kvaser_pci_driver_api);                                         \
+                                                                                                   \
+	static void can_kvaser_pci_config_func_##inst(const struct device *dev)                    \
```

Comment:
we may add BUILD_ASSERT() that dynamic IRQ allocation is not possible

Reply:
Thanks. I will look into that in a follow-up PR.